rdmd.d

```diff
@@ -916,3 +882,74 @@ void yap(size_t line = __LINE__, T...)(auto ref T stuff)
     debug stderr.writeln(line, ": ", stuff);
     else stderr.writeln(stuff);
 }
+
+/**
+Used to wrap filesystem operations that should also be logged with the
+yap function. These functions allow the the filenane to be given once
+so the log statement always matches the operation. Using it also guarantees
+you won't forget to include a `yap` alongside any file operation you want
+to be logged.
+*/
+static struct Yap
+{
+    static import std.file;
+
+    static auto opDispatch(string func, T...)(T args)
+    {
+        static if (func.endsWith("IfLive"))
+        {
+            enum fileFunc = func[0 .. $ - "IfLive".length];
+            enum skipOnDryRun = true;
+        }
+        else
+        {
+            enum fileFunc = func;
+            enum skipOnDryRun = false;
+        }
+
+        static if ([fileFunc].findAmong(["exists", "timeLastModified", "isFile"]))
+            yap("stat ", args[0]);
+        else static if (fileFunc == "rename")
+            yap("mv ", args[0], " ", args[1]);
+        else static if ([fileFunc].findAmong(["remove", "mkdirRecurse", "rmdirRecurse", "dirEntries", "write"]))
+            yap(fileFunc, " ", args[0]);
+        else static assert(0, "Yap.opDispath has not implemented " ~ fileFunc);
+
+        static if (skipOnDryRun)
+        {
+            if (dryRun)
+                return;
+        }
+        mixin("return std.file." ~ fileFunc ~ "(args);");
+    }
+    // Special case, combine these 2 operations so you don't end up with
+    // multiple yap statements in the log for the same file.
+    static bool existsAsFile(const(char)[] name)
```

Comment:
Please remove.

Reply:
I'm going to have to push back on this one.  This will result in a BUNCH of duplicated stat statements on windows

Reply:
But didn't you want to fix this by combining this into one `stat` call (or equiv.) in a follow-up PR?

Reply:
Executing `exists` and `isFile` in succession on the same file is "functionally" the same as performing one `stat` or `GetFileAttributes` function call.  Logging the same thing twice doesn't help anyone here.  It just adds noise to the log and will make people wonder why there are 2 `stat` log statements for the same file

Reply:
> But didn't you want to fix this by combining this into one stat call (or equiv.) in a follow-up PR?

Yes that's a good point.  However, that change would just add this function bad in, just with a different implementation...as long as you understand that, I"ll remove it or leave it in based on what you think.